api/src/main/java/org/pmiops/workbench/api/ClusterController.java

```diff
@@ -34,117 +42,109 @@ public Cluster apply(org.pmiops.workbench.notebooks.model.Cluster firecloudClust
-    Cluster cluster = TO_ALL_OF_US_CLUSTER.apply(
-        this.notebooksService.getCluster(workspaceNamespace, clusterName));
-    return ResponseEntity.ok(cluster);
+  public ResponseEntity<ClusterListResponse> listClusters() {
+    String project = userProvider.get().getFreeTierBillingProjectName();
+    String clusterName = this.clusterNameForUser();
+
+    org.pmiops.workbench.notebooks.model.Cluster fcCluster;
+    try {
+      fcCluster = this.notebooksService.getCluster(project, clusterName);
+    } catch (NotFoundException e) {
+      fcCluster = this.notebooksService.createCluster(project, clusterName, createFirecloudClusterRequest());
+    }
+    ClusterListResponse resp = new ClusterListResponse();
+    resp.setDefaultCluster(TO_ALL_OF_US_CLUSTER.apply(fcCluster));
+    return ResponseEntity.ok(resp);
   }
 
-
-  @Override
-  public ResponseEntity<ClusterListResponse> listClusters(String labels) {
-    List<org.pmiops.workbench.notebooks.model.Cluster> oldClusters;
-    oldClusters = this.notebooksService.listClusters(labels, /* includeDeleted */ false);
-    ClusterListResponse response = new ClusterListResponse();
-    response.setItems(oldClusters.stream().map(TO_ALL_OF_US_CLUSTER).collect(Collectors.toList()));
-    return ResponseEntity.ok(response);
+  private org.pmiops.workbench.notebooks.model.ClusterRequest createFirecloudClusterRequest() {
+    org.pmiops.workbench.notebooks.model.ClusterRequest firecloudClusterRequest = new org.pmiops.workbench.notebooks.model.ClusterRequest();
+    // TODO(calbach): Configure Jupyter server extensions here.
+    Map<String, String> labels = new HashMap<String, String>();
+    labels.put("all-of-us", "true");
+    labels.put("created-by", userProvider.get().getEmail());
+    firecloudClusterRequest.setLabels(labels);
+    return firecloudClusterRequest;
   }
 
   @Override
-  public ResponseEntity<Void> localizeNotebook(String workspaceNamespace, String workspaceId,
-      List<FileDetail> fileList) {
-    String clusterName = convertClusterName(workspaceId);
-    this.notebooksService.localize(workspaceNamespace, clusterName,
-        convertfileDetailsToMap(fileList));
+  public ResponseEntity<Void> localize(
+      String projectName, String clusterName, ClusterLocalizeRequest body) {
+    String workspaceBucket;
+    try {
+      workspaceBucket = "gs://" + fireCloudService.getWorkspace(body.getWorkspaceNamespace(), body.getWorkspaceId())
+          .getWorkspace()
+          .getBucketName();
+    } catch (ApiException e) {
+      if (e.getCode() == 404) {
+        log.log(Level.INFO, "Firecloud workspace not found", e);
+        throw new NotFoundException(String.format(
+            "workspace %s/%s not found or not accessible",
+            body.getWorkspaceNamespace(), body.getWorkspaceId()));
+      }
+      throw ExceptionUtils.convertFirecloudException(e);
+    }
+
+    // For the common case where the notebook cluster matches the workspace
```

Comment:
What's the case where the notebook cluster *doesn't* match the workspace namespace?

If it's one cluster per project + user then it doesn't seem like namespace should be needed...

Reply:
Oh, and the place where namespace is needed is that we only check for name uniqueness within a single project (I believe?), so technically each person can have a `cancer_workspace` for example, within their project.